### PR TITLE
improve #342

### PR DESCRIPTION
**Pull request by RicterZ**
1. change option from `--regenerate-cbz` to `--regenerate`, to indicate program regenerate pdf or cbz file
2. by default, if a pdf or a cbz file exists, skip following download and generation process
3. furthermore, if a pdf file exists, but someone still want to generate a cbz file, he can use `--cbz --regenerate` (under now program logic)

### CODE REVIEW

File: nhentai/doujinshi.py [hunk 1]
```diff
@@ -72,8 +72,14 @@ def __repr__(self):
     def show(self):
         logger.info(f'Print doujinshi information of {self.id}\n{tabulate(self.table)}')
 
-    def download(self, regenerate_cbz=False, file_type=''):
+    def download(self, skip_exists=True):
         logger.info(f'Starting to download doujinshi: {self.name}')
+
+        base_path = os.path.join(self.downloader.path, self.filename)
+        if (os.path.exists(base_path + '.pdf') or os.path.exists(base_path + '.cbz')) and skip_exists:
+            logger.info(f'Skip download doujinshi because a PDF/CBZ file exists of doujinshi {self.name}')
```
Comment (RicterZ) [inline nhentai/doujinshi.py:80]: In fact, this code will never be reached, will be removed in the future

**Comment by RicterZ** [comment]
@normalizedwater546 Does the current program logic meet your needs?

File: nhentai/doujinshi.py (hunk 1 above)
Comment (normalizedwater546) [reply nhentai/doujinshi.py:80]: I was able to reach this code with the following commands:
```
nhentai --id 530833 --download --format '%t [%i]' --threads=4 --pdf --meta --output="test-out"
nhentai --id 530833 --download --format '%t [%i]' --threads=4 --cbz --meta --output="test-out"
```

**Comment by normalizedwater546** [comment]
- If I created a `.pdf` file, I cannot create a `.cbz` file afterwards (or vice versa). 
- It only works if I create them at the same time `--pdf --cbz`.

```
(.venv) PS G:\Exports\nhentai> nhentai --id 530833 --download --format '%t [%i]' --threads=4 --cbz --meta --output="test-out"
...
[19:56:25] download: Skipped download: test-out\[23.4do (Ichiri)] Daten Keikaku - Mou Hitori no Shugosha Hen Fallen Angel Project ~The Other Guardi… [530833]\051.jpg already exists
[19:56:25] generate_metadata_file: Writing Metadata Info
[19:56:25] generate_doc: Writing CBZ file to path: test-out\[23.4do (Ichiri)] Daten Keikaku - Mou Hitori no Shugosha Hen Fallen Angel Project ~The Other Guardi… [530833].cbz
[19:56:25] generate_doc: Comic Book CBZ file has been written to "test-out\[23.4do (Ichiri)] Daten Keikaku - Mou Hitori no Shugosha Hen Fallen Angel Project ~The Other Guardi… [530833].cbz"
[19:56:25] main: All done.
(.venv) PS G:\Exports\nhentai> nhentai --id 530833 --download --format '%t [%i]' --threads=4 --pdf --meta --output="test-out"
[19:56:30] banner: nHentai ver 0.5.7: あなたも変態。 いいね?
[19:56:30] main: Using mirror: https://nhentai.net
[19:56:30] main: Using viewer template "default"
[19:56:31] check_cookie: Login successfully! Your username: --
[19:56:31] doujinshi_parser: Fetching doujinshi information of id 530833
[19:56:32] download: Starting to download doujinshi: [23.4do (Ichiri)] Daten Keikaku - Mou Hitori no Shugosha Hen | Fallen Angel Project ~The Other Guardian~ [Digital] [English] [Kamijoan]
[19:56:32] download: Skip download doujinshi because a PDF/CBZ file exists of doujinshi [23.4do (Ichiri)] Daten Keikaku - Mou Hitori no Shugosha Hen | Fallen Angel Project ~The Other Guardian~ [Digital] [English] [Kamijoan]     
[19:56:32] main: All done.
```

**Comment by normalizedwater546** [comment]
> Does the current program logic meet your needs?

I actually don't need to create both `pdf` and `cbz`, my main goal was to fix the metadata crash when attempting to download duplicated doujins in `cbz` format. I just encountered other issues while debugging. So yes, the current logic is fine for me.

**Comment by normalizedwater546** [comment]
Thanks for the improvements!

Is the change to auto-generate HTML file intentional?

**Comment by RicterZ** [comment]
Yes, the html file is auto-generate, you can add `--no-html` to avoid it

**Pull request merged**